test/e2e/testsuites/podfailovertest.go

```diff
@@ -0,0 +1,111 @@
+	"sigs.k8s.io/azuredisk-csi-driver/test/e2e/driver"
+)
+
+//  will provision required PV(s), PVC(s) and Pod(s)
+// Pod should successfully be re-scheduled on failover in a cluster with AzDriverNode and AzVolumeAttachment resources
+type PodFailover struct {
+	client                 clientset.Interface
+	CSIDriver              driver.DynamicPVTestDriver
+	AzDiskClientSet        v1alpha1ClientSet.DiskV1alpha1Interface
+	AzNamespace            string
+	Pod                    PodDetails
+	Volume                 VolumeDetails
+	StorageClassParameters map[string]string
+}
+
+func (t *PodFailover) Run(client clientset.Interface, namespace *v1.Namespace, schedulerName string) {
+	t.client = client
+	tStatefulSet, cleanup := t.Pod.SetupStatefulset(client, namespace, t.CSIDriver, schedulerName)
+	for i := range cleanup {
+		i := i
+		defer cleanup[i]()
+	}
+
+	// Get the list of available nodes for scheduling the pod
+	nodes := ListNodeNames(client)
+	if len(nodes) < 2 {
+		ginkgo.Skip("need at least 2 nodes to verify the test case. Current node count is %d", len(nodes))
+	}
+
+	volumeName := t.Volume.VolumeMount.NameGenerate + "1"
+	testAzAtt := SetupTestAzVolumeAttachment(t.AzDiskClientSet, t.AzNamespace, volumeName, nodes[0], 0)
+	defer testAzAtt.Cleanup()
+	_ = testAzAtt.Create()
+
+	err := testAzAtt.WaitForAttach(time.Duration(5) * time.Minute)
+	framework.ExpectNoError(err)
+
+	ginkgo.By("deploying the statefulset")
+	tStatefulSet.Create()
+
+	ginkgo.By("checking that the pod for statefulset is running")
+	tStatefulSet.WaitForPodReady()
+
+	// Make node#0 unschedulable to ensure that pods are scheduled on a different node
+	t.SetNodeUnschedulable(nodes[0], true)        // kubeclt cordon node
+	defer t.SetNodeUnschedulable(nodes[0], false) // defer kubeclt uncordon node
+
+	// Define a new scale for statefulset
+	newScale := &scale.Scale{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      tStatefulSet.statefulset.Name,
+			Namespace: tStatefulSet.namespace.Name,
+		},
+		Spec: scale.ScaleSpec{
+			Replicas: int32(0)}}
+
+	// Scale statefulset to 0
+	_, err = client.AppsV1().StatefulSets(tStatefulSet.namespace.Name).UpdateScale(context.TODO(), tStatefulSet.statefulset.Name, newScale, metav1.UpdateOptions{})
+	framework.ExpectNoError(err)
+
+	ginkgo.By("sleep 240s waiting for statefulset update to complete and disk to detach")
+	time.Sleep(240 * time.Second)
+
+	// Scale the stateful set back to 1 pod
+	newScale.Spec.Replicas = int32(1)
+
+	_, err = client.AppsV1().StatefulSets(tStatefulSet.namespace.Name).UpdateScale(context.TODO(), tStatefulSet.statefulset.Name, newScale, metav1.UpdateOptions{})
+	framework.ExpectNoError(err)
+
+	ginkgo.By("sleep 30s waiting for statefulset update to complete")
+	time.Sleep(30 * time.Second)
+
+	ginkgo.By("checking that the pod for statefulset is running")
+	tStatefulSet.WaitForPodReady()
+}
+
+func (t *PodFailover) SetNodeUnschedulable(nodeName string, unschedulable bool) {
```

Comment:
This function already exists in [testsuits](https://github.com/abhisheksinghbaghel/azuredisk-csi-driver/blob/5422820c62b9160f9e94e846b4903d258981ef34/test/e2e/testsuites/testsuites.go#L852). Please reuse that function.